src/jvm/backtype/storm/security/auth/SimpleTransportPlugin.java

```
@@ -0,0 +1,113 @@
+        //create a framed transport
+        TTransport conn = new TFramedTransport(transport);
+
+        //connect
+        conn.open();
+        LOG.debug("Simple client transport has been established");
+
+        return conn;
+    }
+
+    /**                                                                                                                                                                             
+     * Processor that populate simple transport info into ReqContext, and then invoke a service handler                                                                              
+     */
+    private class SimpleWrapProcessor implements TProcessor {
+        final TProcessor wrapped;
+
+        SimpleWrapProcessor(TProcessor wrapped) {
+            this.wrapped = wrapped;
+        }
+
+        public boolean process(final TProtocol inProt, final TProtocol outProt) throws TException {
+            //populating request context 
+            ReqContext req_context = ReqContext.context();
+
+            TTransport trans = inProt.getTransport();
+            if (trans instanceof TMemoryInputTransport) {
+                try {
+                    req_context.setRemoteAddress(InetAddress.getLocalHost());
+                } catch (UnknownHostException e) {
+                    throw new RuntimeException(e);
+                }                                
+            } else if (trans instanceof TSocket) {
+                TSocket tsocket = (TSocket)trans;
+                //remote address
+                Socket socket = tsocket.getSocket();
+                req_context.setRemoteAddress(socket.getInetAddress());                
+            } 
+
+            //anonymous user
+            req_context.setSubject(null);
+
+            //invoke service handler
+            try {
+                return wrapped.process(inProt, outProt);
+            } catch (RuntimeException ex) {
```

Why are you suppressing exception here? Let the runtime exception bubble up. This is changing the behavior of the wrapped processor.